src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java

```diff
@@ -46,13 +44,26 @@ public void actionPerformed(@NotNull AnActionEvent e) {
         }
         List<BuildFile> buildFileList = getBuildFileList(project);
         if (!buildFileList.isEmpty()) {
-            final String[] projectNames = buildFileToProjectNames(buildFileList);
-            final int ret = Messages.showChooseDialog(project,
+            final List<Map<String, String>> projectDetails = buildFilesToProjectDetails(buildFileList);
+            final String[] projectNames = new String[projectDetails.size()];
+            final String[] projectPaths = new String[projectDetails.size()];
+
+            for (int i = 0; i < projectDetails.size(); i++) {
+                Map<String, String> details = projectDetails.get(i);
+                projectNames[i] = details.get("projectName");
```

Comment:
can we avoid using this indexes and please check whether any Java methods available to get the **projectName** in one array and **projectPath** in another array.

Reply:
Changed to use Java streams